src/main/java/wooteco/subway/admin/controller/PathController.java

```diff
@@ -0,0 +1,43 @@
+package wooteco.subway.admin.controller;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import wooteco.subway.admin.dto.PathResponse;
+import wooteco.subway.admin.dto.error.ErrorResponse;
+import wooteco.subway.admin.exception.LineNotConnectedException;
+import wooteco.subway.admin.exception.OverlappedStationException;
+import wooteco.subway.admin.service.PathService;
+
+import javax.validation.constraints.NotBlank;
+
+import static org.springframework.http.HttpStatus.BAD_REQUEST;
+
+@RestController
+@Validated
+public class PathController {
+    private final PathService pathService;
+
+    public PathController(PathService pathService) {
+        this.pathService = pathService;
+    }
+
+    @GetMapping("/paths")
+    public ResponseEntity<PathResponse> findPath(@RequestParam @NotBlank(message = "시작역이 없습니다!") String source,
+                                                 @RequestParam @NotBlank(message = "끝역이 없습니다!") String target,
+                                                 @RequestParam @NotBlank(message = "검색 기준을 선택해주세요") String type) {
```

Comment:
String type 인데 DISTANCE, DURATION이 아닌 문자열이 들어 올 수도 있습니다. PathSearchType으로 받아주면 어떨까요? 😄

Reply:
PathSearchType에 가중치를 선택하는 로직도 추가했습니다!

좀더 객체지향적인 코드가 된 것 같아서 만족스럽네요! :)

Reply:
가중치 👍 

이 부분은 제가 약간 요구사항을 불명확하게 드렸네요. 😢 
```suggestion
                                                 @RequestParam @NotBlank(message = "검색 기준을 선택해주세요")PathSearchType  type) {
```
String이 아니라 enum 타입으로도 받을 수 있답니다. 😉